### PR TITLE
tuhi: init at 0.5

### PR DESCRIPTION
**Pull request by lammermann**
###### Motivation for this change

package tuhi application to export sketches from wacom tablets.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tuhi/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  src = fetchFromGitHub {
+    owner = "tuhiproject";
+    repo = name;
+    rev = "${version}";
+    sha256 = "17kggm9c423vj7irxx248fjc8sxvkp9w1mgawlx1snrii817p3db";
+  };
+
+  meta = with lib; {
+    description = "DBus daemon to access Wacom SmartPad devices";
+    homepage = https://github.com/tuhiproject/tuhi;
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:55]:
```suggestion
    homepage = "https://github.com/tuhiproject/tuhi";
```

File: pkgs/applications/misc/tuhi/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = [ lammermann ];
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:58]:
```suggestion
    maintainers = with maintainers; [ lammermann ];
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -6573,6 +6573,11 @@
     githubId = 55911173;
     name = "Gwendolyn Quasebarth";
   };
+  lammermann = {
+    email = "k.o.b.e.r@web.de";
+    github = "lammermann";
+    name = "Benjamin Kober";
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:6580]: Different commit: `maintainers: add lammermann`

File: pkgs/applications/misc/tuhi/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  postFixup = ''
+  wrapPythonProgramsIn $out/libexec "$out $pythonPath"
+  '';
+
+  src = fetchFromGitHub {
+    owner = "tuhiproject";
+    repo = name;
+    rev = "${version}";
+    sha256 = "17kggm9c423vj7irxx248fjc8sxvkp9w1mgawlx1snrii817p3db";
+  };
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:51]: This should be under the version attribute.

File: pkgs/applications/misc/tuhi/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    svgwrite pyxdg pycairo pygobject3 setuptools-scm
+  ];
+  strictDeps = false;
+  format = "other";
+  preConfigure = ''
+    substituteInPlace meson_install.sh \
+      --replace "/usr/bin/env sh" "sh"
+  '';
+  postFixup = ''
+  wrapPythonProgramsIn $out/libexec "$out $pythonPath"
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:43]:
```suggestion
    wrapPythonProgramsIn $out/libexec "$out $pythonPath"
```

File: pkgs/applications/misc/tuhi/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+python3Packages.buildPythonApplication rec {
+  name = "tuhi";
+  version = "0.5";
+
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+  nativeBuildInputs = [
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:23]:
```suggestion
  
  nativeBuildInputs = [
```

File: pkgs/applications/misc/tuhi/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  ];
+  buildInputs = [
+    gtk3 gobject-introspection
+    glib
+  ];
+  checkInputs = with python3Packages; [ flake8 pytest ];
+  propagatedBuildInputs = with python3Packages; [
+    svgwrite pyxdg pycairo pygobject3 setuptools-scm
+  ];
+  strictDeps = false;
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:36]:
```suggestion
  
  strictDeps = false;
```

**Comment by lammermann** [comment]
I've seen that some checks failed but I don't no what to do to fix that (My knowledge of nix is still very limited).

My current guess is, that some problem with the `pkgconfig` argument prevents the `makeOverridable` function from working correctly. But I have no clue what to change.

Any advice about what I could do to fix the problems would be very much appreciated.

File: pkgs/applications/misc/tuhi/default.nix [hunk 8]
```diff
@@ -0,0 +1,62 @@
+{ lib
+, pkgconfig
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:2]:
```suggestion
, pkg-config
```

File: pkgs/applications/misc/tuhi/default.nix [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    sha256 = "17kggm9c423vj7irxx248fjc8sxvkp9w1mgawlx1snrii817p3db";
+  };
+
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  nativeBuildInputs = [
+    pkgconfig meson ninja
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:32]:
```suggestion
    pkg-config meson ninja
```

**Review by legendofmiracles — COMMENTED** [review]
The error is happening because pkgconfig is actually called pkg-config

**Comment by lammermann** [comment]
Thank you very much. Now I applied your suggestions.

File: pkgs/applications/misc/tuhi/default.nix [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    gtk3 gobject-introspection
+    glib
+  ];
+  checkInputs = with python3Packages; [ flake8 pytest ];
+  propagatedBuildInputs = with python3Packages; [
+    svgwrite pyxdg pycairo pygobject3 setuptools-scm
+  ];
+
+  strictDeps = false;
+  format = "other";
```
Comment (legendofmiracles) [inline pkgs/applications/misc/tuhi/default.nix:46]: Should be under the version attribute

**Review by legendofmiracles — APPROVED** [review]
Can't test this, but other than this LGTM

If it works for you, I'll merge this.

**Comment by lammermann** [comment]
> 

That's great news. Thank you. 

For me it is working. I can access my tablet with tuhi.

**Pull request merged**